src/frontend/src/lib/canisters/reward.canister.ts

```diff
@@ -0,0 +1,50 @@
+import type {
+	ClaimVipRewardResponse,
+	NewVipRewardResponse,
+	_SERVICE as RewardService,
+	UserData,
+	VipReward
+} from '$declarations/rewards/rewards.did';
+import { idlFactory as idlCertifiedFactoryReward } from '$declarations/rewards/rewards.factory.certified.did';
+import { idlFactory as idlFactoryReward } from '$declarations/rewards/rewards.factory.did';
+import { getAgent } from '$lib/actors/agents.ic';
+import type { CreateCanisterOptions } from '$lib/types/canister';
+import { Canister, createServices } from '@dfinity/utils';
+
+export class RewardCanister extends Canister<RewardService> {
+	static async create({
+		identity,
+		...options
+	}: CreateCanisterOptions<RewardService>): Promise<RewardCanister> {
+		const agent = await getAgent({ identity });
+
+		const { service, certifiedService, canisterId } = createServices<RewardService>({
+			options: {
+				...options,
+				agent
+			},
+			idlFactory: idlFactoryReward,
+			certifiedIdlFactory: idlCertifiedFactoryReward
+		});
+
+		return new RewardCanister(canisterId, service, certifiedService);
+	}
+
+	getUserInfo = (): Promise<UserData> => {
+		const { user_info } = this.caller({ certified: true });
+
+		return user_info();
+	};
+
+	getNewVipReward = (): Promise<NewVipRewardResponse> => {
+		const { new_vip_reward } = this.caller({ certified: true });
```

Comment:
ditto

Reply:
According the comment https://github.com/dfinity/oisy-wallet/pull/4016#discussion_r1890344193 I understand it's expected that this call is strictly certified.